test/NuGet.Core.Tests/NuGet.Packaging.Test/RepositoryMetadataTests.cs

```diff
@@ -0,0 +1,72 @@
+using NuGet.Packaging.Core;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    public class RepositoryMetadataTests
+    {
+        [Fact]
+        public void RepositoryMetadataDefaultConstructor()
+        {
+            var metadata = new RepositoryMetadata();
+
+            Assert.Equal(string.Empty, metadata.Type);
+            Assert.Equal(string.Empty, metadata.Url);
+            Assert.Equal(string.Empty, metadata.Branch);
+            Assert.Equal(string.Empty, metadata.Commit);
+        }
+
+        [Theory]
+        [MemberData(nameof(EqualsRepositoryMetadataData))]
+        public void RepositoryMetadataEquals(RepositoryMetadata metadata1, RepositoryMetadata metadata2, bool equals)
+        {
+            Assert.Equal(equals, metadata1 == metadata2);
+            Assert.NotEqual(equals, metadata1 != metadata2);
+
+            if (metadata1 != null)
+            {
+                Assert.Equal(equals, metadata1.Equals(metadata2));
+            }
+            if (metadata1 != null && metadata2 != null)
+            {
+                Assert.Equal(equals, metadata1.GetHashCode() == metadata2.GetHashCode());
+            }
+        }
+
+        public static TheoryData EqualsRepositoryMetadataData
+        {
+            get
+            {
+                var template = new RepositoryMetadata("type", "url", "branch", "commit");
+
+                return new TheoryData<RepositoryMetadata, RepositoryMetadata, bool>
+                {
+                    { template, template, true },
+                    { new RepositoryMetadata(), new RepositoryMetadata(), true },
+                    { new RepositoryMetadata("type", "url", "branch", "commit"), template, true },
+                    { new RepositoryMetadata("TYPE", "url", "branch", "commit"), template, true },
+                    { new RepositoryMetadata("faketype", "url", "branch", "commit"), template, false },
+                    { new RepositoryMetadata("TYPE", "fakeurl", "branch", "commit"), template, false },
+                    { new RepositoryMetadata("TYPE", "url", "fakebranch", "commit"), template, false },
+                    { new RepositoryMetadata("TYPE", "url", "branch", "fakecommit"), template, false },
+                    { new RepositoryMetadata("TYPE", null, "branch", "fakecommit"), template, false },
+                    { new RepositoryMetadata("TYPE", "url", null, "commit"), template, false },
+                    { new RepositoryMetadata("TYPE", "url", "branch", null), template, false },
+                    { template, new RepositoryMetadata(null, "url", "branch", null), false },
+                    { template, null, false },
+                    { null, template, false },
+                };
+            }
+        }
+
+        [Fact]
+        public void RepositoryMetadataEqualsThrows()
+        {
+            Assert.Throws<NullReferenceException>(() => new RepositoryMetadata(null, "url", "branch", "fakecommit") == new RepositoryMetadata());
```

Comment:
oh no, this is terrible :( 

We don't validate the type null do we? 
I think we can safely make Equals return false here right?
What's the concern?

Reply:
Now we do :)
I also think that we can change that safely.
I haven't seen a usage where `Type` was actually `null`.